### ISSUE TITLE
Difference between operators and quotations

### ISSUE THREAD

**Issue by h3rald** [h3rald/min#114]
Every operator defined with the `operator` symbol is, as a matter of fact, a native symbol: the additional checks and expectations that are built-in with the construct are written in Nim by processing the signature, and the body quotation is evaluated.

This means that symbols defined with `operator`:

* cannot be serialized to JSON, therefore `save-symbol` becomes pretty useless.
* cannot be inspected using `source`.
* are inherently very different from "data" symbols defined with `define`.

This should be better clarified in the docs.

**Comment by Econ441** [comment]
A converter which converts operator signutures to normal symbol definitions can handle:

```
( symbol add-one
  (int :n ==> int :result)
  (n 1 + @result)
) operator
```

to

```
( :n n (int) expect ; <- <in-expectation groups here>
" " :result ; <-  <outs here>
n 1 + @result ; <- <body here>
) :add-one
```

But ... there is no a symbol that does things i explained as far as i know, in min.

**Comment by h3rald** [comment]
Well that would work but it would require some more language addition: expect can be used to construct such symbol BUT at the moment it doesn't support generics. While it would be technically possible to extend it or provide a way to validate generics in native min, it is quite a lot of work now... and in the end I am not sure the serialization of operators is worth the hassle.

Other programming languages normally don't support serialization of functions anyway 😉

**Comment by h3rald** [comment]
Fixed in v0.31.0.
